lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -334,6 +305,47 @@ LogicalResult VariableOp::canonicalize(VariableOp op,
   return success();
 }
 
+SmallVector<MemorySlot> VariableOp::getPromotableSlots() {
+  if (isa<SVModuleOp>(getOperation()->getParentOp()))
+    return {};
+  if (getInitial())
+    return {};
```

Comment:
These two returns can be merged 
```suggestion
  if (isa<SVModuleOp>(getOperation()->getParentOp()) || getInitial())
    return {};
```